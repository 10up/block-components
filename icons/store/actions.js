/**
 * Returns an action object used in signalling that new block styles have been added.
 *
 * @param {object} iconSet icon set.
 *
 * @returns {object} Action object.
 */
export function registerIconSet(iconSet) {
	return {
		type: 'REGISTER_ICON_SET',
		iconSet,
	};
}

/**
 * Returns an action object used in signalling that block styles have been removed.
 *
 * @param {string} name  Icon Set name.
 *
 * @returns {object} Action object.
 */
export function removeIconSet(name) {
	return {
		type: 'REMOVE_ICON_SET',
		name,
	};
}
