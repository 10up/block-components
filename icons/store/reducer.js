/**
 * WordPress dependencies
 */
import { combineReducers } from '@wordpress/data';

/**
 * Reducer managing the block style variations.
 *
 * @param {object} state  Current state.
 * @param {object} action Dispatched action.
 *
 * @returns {object} Updated state.
 */
export function iconSets(state = { iconSets: {} }, action) {
	switch (action.type) {
		case 'REGISTER_ICON_SET':
			return {
				...state,
				iconSets: {
					...state.iconSets,
					[action.iconSet.name]: action.iconSet,
				},
			};
		case 'REMOVE_ICON_SET':
			// eslint-disable-next-line no-prototype-builtins
			if (state.iconSet.hasOwnProperty(action.name)) {
				const newState = { ...state };
				delete newState.iconSet[action.name];
				return newState;
			}
			return state;
		default:
			return state;
	}
}

export default combineReducers({
	iconSets,
});
