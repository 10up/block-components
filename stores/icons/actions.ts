import { IconSet, ActionTypes } from "./types";

/**
 * Returns an action object used in signalling that new block styles have been added.
 */
export function registerIconSet(iconSet: IconSet) {
	return {
		type: ActionTypes.REGISTER_ICON_SET,
		iconSet,
	};
}

/**
 * Returns an action object used in signalling that block styles have been removed.
 */
export function removeIconSet(name: string) {
	return {
		type: ActionTypes.REMOVE_ICON_SET,
		name,
	};
}
