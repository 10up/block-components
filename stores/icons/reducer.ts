/* eslint-disable default-param-last */

import { ActionTypes } from './types';

/**
 * Reducer managing the block style variations.
 *
 * @param {object} state  Current state.
 * @param {object} action Dispatched action.
 *
 * @returns {object} Updated state.
 */
export default function reducer(state = { iconSets: {} }, action) {
	switch (action.type) {
		case ActionTypes.REGISTER_ICON_SET:
			return {
				...state,
				iconSets: {
					...state.iconSets,
					[action.iconSet.name]: action.iconSet,
				},
			};
		case ActionTypes.REMOVE_ICON_SET:
			// eslint-disable-next-line no-prototype-builtins
			if (state.iconSets.hasOwnProperty(action.name)) {
				const newState = { ...state };
				delete newState.iconSets[action.name];
				return newState;
			}
			return state;
		default:
			return state;
	}
}
