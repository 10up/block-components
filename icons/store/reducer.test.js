import reducer from './reducer';
import { registerIconSet } from './actions';

describe( 'icon store | reducer', () => {
    test('iconSets', () => {
        const state = {
            iconSets: {
                iconSets: {}
            }
        };

        const newIconsSetData = {
            name: 'example/theme',
            label: "Example",
            icons: [
                {
                    source: 'test-string',
                    name: "search",
                    label: "Search"
                }
            ]
        };

        const registerIconAction = registerIconSet(newIconsSetData);
    
        const newState = reducer(state, registerIconAction);
        expect(newState).toMatchSnapshot();
    })
} )