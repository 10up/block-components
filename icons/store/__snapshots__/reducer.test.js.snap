// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon store | reducer iconSets 1`] = `
Object {
  "iconSets": Object {
    "example/theme": Object {
      "icons": Array [
        Object {
          "label": "Search",
          "name": "search",
          "source": "test-string",
        },
      ],
      "label": "Example",
      "name": "example/theme",
    },
  },
}
`;
